### PR TITLE
Add support of complex dtypes for `sinc`

### LINKED ISSUE #2645: cp.sinc fails for complex input
Unlike the numpy version, `cupy.sinc` fails on complex inputs:

```python
>>> np.sinc(1j)
(3.676077910374978+0j)
>>> cp.sinc(1j)
TypeError                                 Traceback (most recent call last)
----> 1 cp.sinc(1j)
cupy/core/_kernel.pyx in cupy.core._kernel.ufunc.__call__()
cupy/core/_kernel.pyx in cupy.core._kernel._guess_routine()
TypeError: Wrong type ((<class 'numpy.complex128'>,)) of arguments for cupy_sinc
```

Either the docs should be updated or the underlying implementation should be fixed.

```python
>>> cp.show_config()
CuPy Version          : 6.0.0
CUDA Root             : /usr/local/cuda
CUDA Build Version    : 10000
CUDA Driver Version   : 10010
CUDA Runtime Version  : 10000
cuDNN Build Version   : 7301
cuDNN Version         : 7604
NCCL Build Version    : 1000
NCCL Runtime Version  : (unknown)
```

### PR DESCRIPTION
**Pull request by emcastillo**
Closes #2645 

`sinpi` is a cuda only function that does not have an equivalent in thrust for complex numbers.
Note that the original numpy implementation uses the product of pi instead of an optimized sin.

Question:
Should we split the routines using a dispatcher, as the ufuncs doesnt seem to support overloading functions, one for complex and one for non-complex numbers?

### CODE REVIEW

**Comment by asi1024** [comment]
(It is not documented explicitly, but) `core.create_ufunc` can create a `ufunc` that switches their behavior by input type. Here is an example.

https://github.com/cupy/cupy/blob/59e6c2b2e0c722b09c7a7af13f908942ef7806cc/cupy/math/arithmetic.py#L8-L21

**Comment by emcastillo** [comment]
Oh! this is pretty useful
Thanks a lot, I will refactor it using that approach.

File: cupy/math/special.py [hunk 1]
```diff
@@ -12,7 +12,12 @@
 
 
 sinc = core.create_ufunc(
-    'cupy_sinc', ('e->e', 'f->f', 'd->d'),
+    'cupy_sinc',
+    ('e->e', 'f->f', 'd->',
```
Comment (asi1024) [inline cupy/math/special.py]:
```suggestion
    ('e->e', 'f->f', 'd->d',
```

**Comment by asi1024** [comment]
Jenkins, test this please.

**Comment by asi1024** [comment]
LGTM.

**Pull request merged**